python/pyhive/sqlalchemy_hive.py

```diff
@@ -377,7 +377,17 @@ def get_table_names(self, connection, schema=None, **kw):
         query = 'SHOW TABLES'
         if schema:
             query += ' IN ' + self.identifier_preparer.quote_identifier(schema)
-        return [row[1] for row in connection.execute(text(query))]
+
+        table_names = []
+
+        for row in connection.execute(text(query)):
+            # Hive returns 1 columns
+            if len(row) == 1:
+                table_names.append(row[0])
+            # Kyuubi returns 3 columns
```

Comment:
actually, for SQL cases, Kyuubi's behavior totally depends on the computing engine
```suggestion
            # Spark SQL returns 3 columns
```